### PR TITLE
[improvement](group commit) Add an option of enable relay wal on all regression test

### PR DESCRIPTION
**Pull request by hust-hhb**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

File: be/src/olap/wal_manager.cpp [hunk 1]
```diff
@@ -409,4 +416,80 @@ Status WalManager::get_wal_column_index(int64_t wal_id, std::vector<size_t>& col
...
+    auto cv_it = _wal_cv_map.find(wal_id);
+    if (cv_it == _wal_cv_map.end()) {
+        return Status::InternalError("cannot find txn {} in wal_cv_map", wal_id);
+    }
+    cv = cv_it->second;
+    return Status::OK();
+}
+
+bool WalManager::find_wal_path(int64_t wal_id) {
+    std::shared_lock rdlock(_wal_lock);
```
Comment (mymeiyi) [inline be/src/olap/wal_manager.cpp:484]: Duplicated with `get_wal_path`

File: be/src/olap/wal_manager.h [hunk 2]
```diff
@@ -69,6 +69,14 @@ class WalManager {
...
     void erase_wal_column_index(int64_t wal_id);
     Status get_wal_column_index(int64_t wal_id, std::vector<size_t>& column_index);
+    Status add_wal_cv_map(int64_t wal_id, std::shared_ptr<std::mutex> lock,
+                          std::shared_ptr<std::condition_variable> cv);
+    Status erase_wal_cv_map(int64_t wal_id);
+    Status get_lock_and_cv(int64_t wal_id, std::shared_ptr<std::mutex>& lock,
+                           std::shared_ptr<std::condition_variable>& cv);
+    Status wait_relay_wal_finish(int64_t wal_id);
+    bool find_wal_path(int64_t wal_id);
+    Status notify(int64_t wal_id);
```
Comment (mymeiyi) [inline be/src/olap/wal_manager.h:79]: `notify` is a bad name

File: be/src/olap/wal_manager.h [hunk 3]
```diff
@@ -86,6 +94,9 @@ class WalManager {
     std::atomic<bool> _stop;
     std::shared_mutex _wal_column_id_map_lock;
     std::unordered_map<int64_t, std::vector<size_t>&> _wal_column_id_map;
+    std::shared_mutex _wal_cv_lock;
+    std::unordered_map<int64_t, std::shared_ptr<std::mutex>> _wal_lock_map;
+    std::unordered_map<int64_t, std::shared_ptr<std::condition_variable>> _wal_cv_map;
```
Comment (mymeiyi) [inline be/src/olap/wal_manager.h:99]: why not add lock and cv in a map

File: be/src/runtime/runtime_state.h [hunk 4]
```diff
@@ -257,6 +257,10 @@ class RuntimeState {
 
     int64_t wal_id() { return _wal_id; }
 
+    void set_txn_id(int64_t txn_id) { _txn_id = txn_id; }
+
```
Comment (mymeiyi) [inline be/src/runtime/runtime_state.h:261]: the txn_id is unused?

File: be/src/vec/sink/writer/vtablet_writer.cpp [hunk 5]
```diff
@@ -1660,6 +1660,15 @@ Status VTabletWriter::append_block(doris::vectorized::Block& input_block) {
                     block.get(), filter_col, block->columns()));
         }
     }
+#ifndef BE_TEST
```
Comment (mymeiyi) [inline be/src/vec/sink/writer/vtablet_writer.cpp:1663]: can we move it to `GroupCommitBlockSink`?

File: regression-test/pipeline/p0/conf/be.conf [hunk 6]
```diff
@@ -77,3 +77,9 @@ enable_feature_binlog=true
 max_sys_mem_available_low_water_mark_bytes=69206016
 user_files_secure_path=/
 enable_debug_points=true
+group_commit_wal_path=/mnt/ssd01/cluster_storage/doris.SSD/P0/cluster1/wal1;/mnt/ssd01/cluster_storage/doris.SSD/P0/cluster1/wal2
```
Comment (mymeiyi) [inline regression-test/pipeline/p0/conf/be.conf:80]: reset this change

**Comment by hust-hhb** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Comment by dataroaring** [comment]
run buildall

**Comment by hust-hhb** [comment]
run buildall

**Review by mymeiyi — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Comment by hust-hhb** [comment]
run buildall

**Pull request merged**